### PR TITLE
Decrease stalebot interval per collab summit discussion

### PR DESCRIPTION
**Pull request by bmuenzenmeyer**
At the Node.js Collab Summit in September, I presented a small quantitative analysis of this repository's growth.

image

Within https://github.com/nodejs/admin/issues/830 a group of us collaborators discussed actions to make the triage experience better here. Among the consensus at the meeting was to emphasize timely engagement here to avoid stale issues from going unanswered. That doesn't respect askers time or triagers time wading through years' old questions.

The goal with this action and more is to level-up the triage and help posture to such a degree that our aggregate triaging efforts can be best funneled toward the issues where posters remain committed to receiving help.

### CODE REVIEW

File: .github/workflows/close-stale-issues.yml [hunk 1]
```diff
@@ -12,25 +12,25 @@ env:
...
+    If you are still experiencing this issue on the latest supported versions of Node.js, please leave a comment.
 # yamllint enable
 
 jobs:
   stale:
     if: github.repository == 'nodejs/help'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/stale@v4
+      - uses: actions/stale@a20b814fb01b71def3bd6f56e7494d667ddf28da #4.1.1
```
Comment (bmuenzenmeyer) [inline .github/workflows/close-stale-issues.yml:26]: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

File: .github/workflows/close-stale-issues.yml [hunk 2]
```diff
@@ -12,25 +12,25 @@ env:
...
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-          # 3 years. this number is chosen to target around 25 initial issues, with then a natural flow as time progresses
-          days-before-stale: 1095
+          days-before-stale: 90
```
Comment (bmuenzenmeyer) [inline .github/workflows/close-stale-issues.yml:29]: open to other numbers...but in total this means that no activity in a THIRD of a year will close the issue. I think that sounds like a fair balance of empathy toward maintainer and asker time/schedules.

During the collab summit I also suggested multiple PRs that go from 3 years, to 2 years, to 1 year, to 6 months, or something like that... but it sounded good on paper and didn't land well with the audience...the only benefit is shortening what will be many notifications. Node.js collaborators are already used to a firehose and should have optimized the experience to their needs.

**Review by bmuenzenmeyer — COMMENTED** [review]
self-review

**Comment by gireeshpunathil** [comment]
the 90 days mark looks very aggressive to me.
 - the voluntary nature of engagement means triagers (and collaborators too) who engage to help progress the issue) might catch attention of the issue at an unpredictably random time frame.
 - potentially transient nature of issues will mean that the `askers` may not respond instantly to a suggestion, but may be after many weeks / months when they notice the issue next time or when they are in a position to apply the change control in their production.

In summary, while 3 years may be a little too long wait period, shortening it aggressively can inadvertently close issues that may be otherwise progressed and resolved. Having said that, an 18 months of staleness looks reasonable to me for abandonment.

As a side note, a more comprehensive way to address this is to institute a process in place that makes sure some regular trigger engagements in the repo, including running workshops to triage down the backlog. Happy to engage and help here!

File: .github/workflows/close-stale-issues.yml [hunk 3]
```diff
@@ -12,25 +12,25 @@ env:
     and it is being closed. If you feel closing this issue is not the
     right thing to do, please leave a comment.
   WARN_MESSAGE: >
-    There has been no activity on this issue for
-    3 years and it may no longer be relevant.
-    It will be closed 1 month after the last non-automated comment.
+    There has been no activity on this issue for 90 days.
```
Comment (gireeshpunathil) [inline .github/workflows/close-stale-issues.yml:15]:
```suggestion
    There has been no activity on this issue for 365 days.
```

**Comment by bmuenzenmeyer** [comment]
hey @gireeshpunathil  - i gave you a very well deserved shout out in my slides - 
image

and appreciate you weighing in on what a good balance should be.



> As a side note, a more comprehensive way to address this is to institute a process in place that makes sure some regular trigger engagements in the repo, including running workshops to triage down the backlog. Happy to engage and help here!

Totally agree! If you check out https://github.com/nodejs/admin/issues/830 you'll see that this is just one of 4 or 5 things we want to do to improve the triage experience - two of which is creating are creating some triage guidance and regularly meeting as a triage team to help one another. I intend to find the time to organize that in the coming weeks and would be thrilled if you could make it.

**Review by mcollina — APPROVED** [review]
lgtm

I agree with this change. The tracker is out of control.

**Comment by bmuenzenmeyer** [comment]
Perhaps as a compromise between the two expressed opinions (3 months, 18 months...) we could experiment with `9`. That about splits the difference.

**Comment by preveen-stack** [comment]
A graceful closure would be that done by the Author, who in many cases are not accessible for requests for more details.

Most issues are help requests which may not be in scope of nodejs development as such

 If anybody who is well versed with nodejs internals can afford to spare the time to browse through the issues once a month or so, I think that will result in resolution of many issues that needs expert attention which @nodejs/issue-triage members are unable move closer to closure.

Making the closure deadline more aggressive will probably result in seeing number of issues reduced, but that is not serving the purpose of this repo.

I would suggest that automated tool should ping the author 3 times with 3 months duration between each ping, and if not response received we can go for automated closure

**Comment by bmuenzenmeyer** [comment]
Seeking consensus on:

- 11 month staleness
- 1 month closure after that

Here's a survey of what would be closed with this set of values:

https://github.com/nodejs/help/issues?q=is%3Aissue+is%3Aopen+updated%3A%3C2022-10-20

[387 of 653 issues with no activity for a year](https://github.com/nodejs/help/issues?q=is%3Aissue+is%3Aopen+updated%3A%3C2022-10-20) - or 60% closed.

These aren't gone forever, they can be re-opened with a simple request.

**Review by gireeshpunathil — APPROVED** [review]

**Comment by gireeshpunathil** [comment]
I am good with approach mentioned in https://github.com/nodejs/help/pull/4270#issuecomment-1773172948 . @mcollina - are you good too?

**Comment by bmuenzenmeyer** [comment]
let's do it - this strikes the right balance
